org.eclipse.acute/src/org/eclipse/acute/dotnetrun/DotnetRunTab.java

```diff
@@ -0,0 +1,123 @@
+package org.eclipse.acute.dotnetrun;
+
+import static org.eclipse.swt.events.SelectionListener.widgetSelectedAdapter;
+
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;;
+
+public class DotnetRunTab extends AbstractLaunchConfigurationTab {
+
+	private Text pathText;
+	private Text argumentsText;
+	private Button buildCheckBoxButton;
+
+	@Override
+	public void createControl(Composite parent) {
+		Composite container = new Group(parent, SWT.BORDER);
+		setControl(container);
+		GridLayoutFactory.swtDefaults().numColumns(4).applyTo(container);
+
+		Label locationLabel = new Label(container, SWT.NONE);
+		locationLabel.setText("Location:");
+		locationLabel.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false, 2, 1));
+
+		pathText = new Text(container, SWT.BORDER);
+		pathText.addModifyListener(e -> {
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		});
+
+		Button browseButton = new Button(container, SWT.NONE);
+		browseButton.setText("Browse...");
+		browseButton.addSelectionListener(widgetSelectedAdapter(e -> {
+			DirectoryDialog dialog = new DirectoryDialog(browseButton.getShell());
+			String path = dialog.open();
+			if (path != null) {
+				pathText.setText(path.toString());
+				setDirty(true);
+				updateLaunchConfigurationDialog();
+			}
+		}));
+
+		GridDataFactory.fillDefaults().grab(true, false).applyTo(pathText);
+
+		Label argumentLabel = new Label(container, SWT.NONE);
+		argumentLabel.setText("Arguments:");
+		argumentLabel.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false, 2, 1));
+
+		argumentsText = new Text(container, SWT.BORDER);
+		argumentsText.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false, 2, 1));
+		argumentsText.addModifyListener(e -> {
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		});
+
+		new Label(container, SWT.NONE);
```

Comment:
With a grid, you can create a single "filler" Label like this and set it a GridData with an horizontal span of 4 to avoid repeating the same line.